src/qp_block_data.jl

```diff
@@ -0,0 +1,411 @@
+# Copyright (c) 2013: Iain Dunning, Miles Lubin, and contributors
+#
+# Use of this source code is governed by an MIT-style license that can be found
+# in the LICENSE.md file or at https://opensource.org/licenses/MIT.
+
+# Taken from `Ipopt.jl/src/utils.jl`
+
+# We don't support parameters in `Manopt.Optimizer` yet
+_is_parameter(v) = false
+
+@enum(
+    _FunctionType,
+    _kFunctionTypeVariableIndex,
+    _kFunctionTypeScalarAffine,
+    _kFunctionTypeScalarQuadratic,
+)
+
+function _function_type_to_set(::Type{T}, k::_FunctionType) where {T}
+    if k == _kFunctionTypeVariableIndex
+        return MOI.VariableIndex
+    elseif k == _kFunctionTypeScalarAffine
+        return MOI.ScalarAffineFunction{T}
+    else
+        @assert k == _kFunctionTypeScalarQuadratic
+        return MOI.ScalarQuadraticFunction{T}
+    end
+end
+
+_function_info(::MOI.VariableIndex) = _kFunctionTypeVariableIndex
+_function_info(::MOI.ScalarAffineFunction) = _kFunctionTypeScalarAffine
+_function_info(::MOI.ScalarQuadraticFunction) = _kFunctionTypeScalarQuadratic
+
+@enum(_BoundType, _kBoundTypeLessThan, _kBoundTypeGreaterThan, _kBoundTypeEqualTo,)
+
+_set_info(s::MOI.LessThan) = _kBoundTypeLessThan, -Inf, s.upper
+_set_info(s::MOI.GreaterThan) = _kBoundTypeGreaterThan, s.lower, Inf
+_set_info(s::MOI.EqualTo) = _kBoundTypeEqualTo, s.value, s.value
+
+function _bound_type_to_set(::Type{T}, k::_BoundType) where {T}
+    if k == _kBoundTypeEqualTo
+        return MOI.EqualTo{T}
+    elseif k == _kBoundTypeLessThan
+        return MOI.LessThan{T}
+    else
+        @assert k == _kBoundTypeGreaterThan
+        return MOI.GreaterThan{T}
+    end
+end
+
+mutable struct QPBlockData{T}
```

Comment:
this should be documented as well. For me neither Q nor P not Block nor Data mean anything in our context (yet) – unless this file is 1-1 from the other package – then it could maybe stay as is, but I still would not be so happy to have a completely undocumented file in a codebase, where I would probably be the first to be asked questions about (and I have nearly no idea what this part here does).

Reply:
Removed